go/obscuronode/enclave/sql/edbattestation.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+	return getCertificate(host, config, enclave.VerifyRemoteReport)
+}
+
+// InsecureGetCertificate gets the TLS certificate from the server in PEM format, but does not perform remote attestation.
+func InsecureGetCertificate(host string) ([]*pem.Block, error) {
+	certs, _, err := getCertificate(host, nil, nil)
+	return certs, err
+}
+
+type verifyFunc func([]byte) (attestation.Report, error)
+
+func getCertificate(host string, config []byte, verifyRemoteReport verifyFunc) ([]*pem.Block, tcbstatus.Status, error) {
+	// we skip verify because we have the security of the attestation
+	cert, quote, err := httpGetCertQuote(&tls.Config{InsecureSkipVerify: true}, host, "quote") //nolint:gosec
+	if err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+
+	var certs []*pem.Block
+	block, rest := pem.Decode([]byte(cert))
+	if block == nil {
+		return nil, tcbstatus.Unknown, errors.New("could not parse certificate")
+	}
+	certs = append(certs, block)
+
+	// If we get more than one certificate, append it to the slice
+	for len(rest) > 0 {
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return nil, tcbstatus.Unknown, errors.New("could not parse certificate chain")
+		}
+		certs = append(certs, block)
+	}
+
+	if verifyRemoteReport == nil {
+		return certs, tcbstatus.Unknown, nil
+	}
+
+	if len(quote) == 0 {
+		return nil, tcbstatus.Unknown, ErrEmptyQuote
+	}
+
+	report, verifyErr := verifyRemoteReport(quote)
+	if verifyErr != nil && !errors.Is(verifyErr, attestation.ErrTCBLevelInvalid) {
+		return nil, tcbstatus.Unknown, verifyErr
+	}
+
+	// Use Root CA (last entry in certs) for attestation
+	certRaw := certs[len(certs)-1].Bytes
+
+	if err := verifyReport(report, certRaw, config); err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+
+	return certs, report.TCBStatus, verifyErr
+}
+
+func verifyReport(report attestation.Report, cert []byte, config []byte) error {
+	hash := sha256.Sum256(cert)
+	if !bytes.Equal(report.Data[:len(hash)], hash[:]) {
+		return errors.New("report data does not match the certificate's hash")
+	}
+
+	var cfg struct {
+		SecurityVersion uint
+		UniqueID        string
+		SignerID        string
+		ProductID       uint16
+		Debug           bool
+	}
+	if err := json.Unmarshal(config, &cfg); err != nil {
+		return err
+	}
+	if cfg.UniqueID == "" {
+		if cfg.SecurityVersion == 0 {
+			return errors.New("missing securityVersion in config")
+		}
+		if cfg.ProductID == 0 {
+			return errors.New("missing productID in config")
+		}
+	}
+
+	if cfg.SecurityVersion != 0 && report.SecurityVersion < cfg.SecurityVersion {
+		return errors.New("invalid security version")
+	}
+	if cfg.ProductID != 0 && binary.LittleEndian.Uint16(report.ProductID) != cfg.ProductID {
+		return errors.New("invalid product")
+	}
+	if report.Debug && !cfg.Debug {
+		return errors.New("debug enclave not allowed")
+	}
+	if err := verifyID(cfg.UniqueID, report.UniqueID, "unqiueID"); err != nil {
```

Comment:
@BedrockSquirrel Can you pull `uniqueID` and `signerID` out as constants?

Reply:
They're just descriptions of the ID field so you can tell which one failed if it failed, I think to give them a constant would make it seem like something meaningful to configure. That's why the typo never mattered here.